### ISSUE TITLE
Revalidate recurrent_ppo_flickering_cartpole under global-norm clip semantics (post-#300)

### ISSUE THREAD

**Issue by rjwalters** [rjwalters/thrust#301]
PR #300 (issue #299) migrated `recurrent_ppo_flickering_cartpole` off its optimizer-level `GradientClippingConfig::Norm(0.5)` workaround onto `PPOConfig::max_grad_norm(0.5)`, which the trainers now apply as a **global** L2-norm clip (`train::grad_clip::clip_grads_by_global_norm`).

These are not numerically identical:
- **Old (workaround)**: Burn's per-tensor norm clip — each parameter tensor independently capped at 0.5; total step norm can reach `sqrt(K) * 0.5` for K tensors, and the joint gradient direction is distorted.
- **New (config field)**: global-norm clip — total step norm capped at 0.5, direction preserved (SB3/PyTorch `clip_grad_norm_` semantics).

The global clip is never *looser* than the per-tensor clip at the same cap, so the load-bearing safety property (bounding value-loss spikes through the shared LSTM trunk, see PR #298) is preserved or strengthened. The residual risk is the opposite direction: strictly tighter clipping could slow learning enough to affect the documented outcome (LSTM clears the 195 solved bar within 500k steps; #298 measured LSTM 467.6 final / 483.9 peak vs MLP 163.0 / 177.4).

The example was compile-checked but **not retrained** in PR #300 (2 arms x 500k CPU steps).

## Task

- Re-run `cargo run --example recurrent_ppo_flickering_cartpole --features training --release` (both arms, default 500k steps, seed 0).
- Confirm the LSTM arm still clears the >195 solved bar and the LSTM-vs-MLP contrast remains decisive and correctly signed.
- Update the reference numbers in the issue/PR record (and any tuning comments in the example that no longer hold, e.g. the LR-annealing rationale) if the trajectories shifted materially.
- If the LSTM arm no longer clears the bar, tune under the new (intended) clip semantics rather than reverting to the optimizer-level workaround.

Discovered during Judge review of PR #300.

**Comment by rjwalters** [comment]
## Curation (orchestrator, fast-track — scope defined by the #300 judge)

**Task:** rerun the two 500k-step arms of `recurrent_ppo_flickering_cartpole` on current main, where clipping is now `clip_grads_by_global_norm(0.5)` via `PPOConfig` instead of the old per-tensor `GradientClipping::Norm(0.5)` workaround, and refresh the reference numbers.

### Acceptance criteria
1. Run both arms (LSTM via RecurrentPPOTrainer, MLP baseline) at 500k steps, seed 0, release build — no code changes to trainers or env.
2. Report final/best mean returns for both arms on issue #287-style terms and update any reference numbers recorded in the example's doc comments (if none are baked in, record results on this issue and in a brief PR body note).
3. **Honest-reporting guardrail:** if the LSTM no longer clears the 195 bar or the memory contrast collapses under global-norm semantics, do NOT tune hyperparameters to force it. Report the numbers, diagnose briefly (e.g. effective clip tightness vs per-tensor), and escalate on this issue.
4. If numbers hold (expected): PR is docs/comment-only or no-op + issue comment; if doc comments change, standard bar (fmt, clippy, doc build).

**Comment by rjwalters** [comment]
## Revalidation PASSED under global-norm clip semantics

Reran both arms of `recurrent_ppo_flickering_cartpole` on current main (`ebcb7ae`, includes PR #300's `PPOConfig::max_grad_norm(0.5)` -> `clip_grads_by_global_norm` migration). Release build, NdArray CPU backend, seed 0, default 500k steps per arm, flicker p=0.5.

| Arm | final mean (last <=100 eps) | best mean | #298 reference (per-tensor clip) |
|---|---|---|---|
| LSTM (RecurrentPPOTrainer) | **474.6** | **483.9** | 467.6 / 483.9 |
| MLP baseline (PPOTrainerBurn) | **135.0** | **176.0** | 163.0 / 177.4 |

Verbatim output:

```
[lstm] done in 270.0s — final mean return 474.6 (best 483.9)
[mlp ] done in 24.4s — final mean return 135.0 (best 176.0)
LSTM CLEARED solved bar (195) within budget (peak 483.9); final 474.6.
Memory advantage (peak): LSTM 483.9 vs feedforward 176.0 — gap 307.9 (64% of LSTM).
```

### Assessment vs acceptance criteria

1. **LSTM still clears the >195 solved bar**: peak 483.9, final 474.6 — both comfortably above; final is actually slightly *better* than the #298 reference (467.6). The strictly-tighter global clip did not slow learning materially.
2. **Memory contrast remains decisive and correctly signed**: LSTM peak 483.9 vs MLP peak 176.0 (gap 307.9, 64% of LSTM peak). The MLP arm's final (135.0 vs 163.0) differs within normal end-of-run oscillation at episode granularity; its peak (176.0 vs 177.4) matches the reference closely, and it remains below the solved bar as expected for a memoryless policy on the flickering POMDP.
3. **No source changes needed**: the example bakes no reference numbers into its doc comments; the gradient-clipping comments already describe the global-norm semantics (updated in PR #300); the LR-annealing rationale still holds (final policy stays above the bar). No hyperparameters were touched.

Closing: revalidation passed, docs-only diff was empty, results recorded here per the curation acceptance criteria.
